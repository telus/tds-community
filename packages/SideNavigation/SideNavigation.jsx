import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import safeRest from '@tds/shared-safe-rest'
import { colorGainsboro } from '@tds/core-colours'
import Box from '@tds/core-box'
import Text from '@tds/core-text'

import Link from './Link/Link'
import SubMenu from './SubMenu/SubMenu'

const DivContainer = styled.div({
  position: 'relative',
  height: '100%',
})

const topPosition = {
  position: 'relative',
  maxWidth: 'inherit',
  width: '100%',
  overflowY: 'auto',
}

const fixedPosition = {
  position: 'fixed',
  maxWidth: 'inherit',
  top: '0px',
  width: 'inherit',
  clear: 'both',
}

const bottomPosition = {
  position: 'absolute',
  bottom: '0px',
}

const fixedOverflow = {
  overflowY: 'auto',
  bottom: '0px',
}

const NavContainer = styled.div(props => ({
  ...(props.variant === 'top' && topPosition),
  ...(props.variant === 'bottom' && bottomPosition),
  ...(props.variant === 'fixed' && fixedPosition),
  ...(props.variant === 'fixedOverflow' && {
    ...fixedPosition,
    ...fixedOverflow,
  }),
}))

const StyledUl = styled.ul({
  listStyle: 'none',
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'space-around',
  margin: '0',
  borderTop: `1px solid ${colorGainsboro}`,
})

const StyledLi = styled.li({
  borderBottom: `1px solid ${colorGainsboro}`,
  alignItems: 'center',
  lineHeight: '0',
})

/**
 * The SideNavigation component is used in conjuntion with a large amount of educational / informational content,
 * allowing the user to navigate between options frequently and efficiently.
 * @version ./package.json
 */
class SideNavigation extends Component {
  constructor(props) {
    super(props)
    this.state = {
      open: [],
      variant: 'top',
      accordion: this.props.accordion,
    }
    this.adjustWidth = this.adjustWidth.bind(this)
    this.removeEventListeners = this.removeEventListeners.bind(this)
    this._sideNav = React.createRef()
  }

  componentDidMount() {
    this.checkOffset()
    window.addEventListener('scroll', this.checkOffset)
    window.addEventListener('click', this.checkOffset)
    this.adjustWidth()
    window.addEventListener('resize', this.adjustWidth)
    this.checkActiveState()
  }

  componentWillUnmount() {
    this.removeEventListeners()
  }

  adjustWidth() {
    const parentWidth = this._sideNavContainer.offsetWidth
    const sideNav = this._sideNav.current
    sideNav.style.width = `${parentWidth}px`
  }

  toggleSubMenu = id => {
    if (this.checkAccordion(id, this.state.accordion)) {
      const array = [...this.state.open]
      const index = array.indexOf(id)
      if (index !== -1) {
        array.splice(index, 1)
        this.setState({ open: array })
      }
    } else if (this.state.accordion) {
      this.setState({ open: [id] })
    } else {
      this.setState({ open: [...this.state.open, id] })
    }
  }

  removeEventListeners() {
    window.removeEventListener('scroll', this.checkOffset)
    window.removeEventListener('click', this.checkOffset)
    window.removeEventListener('resize', this.adjustWidth)
  }

  checkOffset = () => {
    const sideNavRect = this._sideNav.current.getBoundingClientRect()
    const containerRect = this._sideNavContainer.getBoundingClientRect()

    if (
      (sideNavRect.top >= 0 && containerRect.top >= 0) ||
      sideNavRect.height > containerRect.height
    ) {
      this.setState({ variant: 'top' })
    } else if (
      this.checkOverflow(this._sideNav.current) &&
      sideNavRect.bottom <= containerRect.bottom &&
      this.state.variant !== 'bottom' &&
      sideNavRect.height < containerRect.height
    ) {
      this.setState({ variant: 'fixedOverflow' })
    } else if (
      ((sideNavRect.top < 0 &&
        containerRect.top < 0 &&
        sideNavRect.bottom <= containerRect.bottom &&
        this.state.variant !== 'bottom') ||
        (sideNavRect.top > 0 && containerRect.top <= 0)) &&
      sideNavRect.height < containerRect.height
    ) {
      this.setState({ variant: 'fixed' })
    } else if (
      (sideNavRect.bottom > containerRect.bottom || sideNavRect.bottom <= 0) &&
      sideNavRect.height <= containerRect.height &&
      this.state.variant !== 'top'
    ) {
      this.setState({ variant: 'bottom' })
    }
  }

  checkAccordion = id => {
    return this.state.open.some(el => {
      return el === id
    })
  }

  checkOverflow = element => {
    return element.scrollHeight > window.innerHeight
  }

  checkActiveState = () => {
    React.Children.map(this.props.children, (child, index) => {
      const id = `TDS-SideNavigation-${index}`
      if (!('href' in child.props) && child.props.active) {
        this.toggleSubMenu(id)
      }
    })
  }

  render() {
    const { children, verticalSpacing, accordion, category, ...rest } = this.props
    const { variant } = this.state

    return (
      <DivContainer
        {...safeRest(rest)}
        ref={c => {
          this._sideNavContainer = c
        }}
      >
        <NavContainer ref={this._sideNav} variant={variant}>
          <Box vertical={variant === 'bottom' || variant === 'fixed' ? undefined : verticalSpacing}>
            {category && (
              <Box vertical={3} horizontal={3}>
                <Text size="large" bold>
                  {category}
                </Text>
              </Box>
            )}
            <StyledUl>
              {React.Children.map(children, (child, index) => {
                let options = {}
                const id = `TDS-SideNavigation-${index}`
                // check if href is in props to figure out if child is SubMenu or Link
                if (!('href' in child.props)) {
                  options = {
                    handleToggleSubMenu: this.toggleSubMenu,
                    isOpen: this.checkAccordion(id),
                    id,
                  }
                }
                return <StyledLi>{React.cloneElement(child, options)}</StyledLi>
              })}
            </StyledUl>
          </Box>
        </NavContainer>
      </DivContainer>
    )
  }
}

SideNavigation.propTypes = {
  /**
   * Specifies the label at the top of the Side Navigation.
   */
  category: PropTypes.string,
  /**
   * Specifies the links and sub-menus required in the Side Navigation.
   */
  children: PropTypes.node.isRequired,
  /**
   * Indent content from the container's top edge by applying padding.
   */
  verticalSpacing: PropTypes.oneOf([1, 2, 3, 4, 5, 6, 7, 8]),
  /**
   * Identifies if only one `SideNavigation.SubMenu` should be open at a time.
   */
  accordion: PropTypes.bool,
}

SideNavigation.defaultProps = {
  verticalSpacing: undefined,
  accordion: true,
  category: undefined,
}

SideNavigation.Link = Link
SideNavigation.SubMenu = SubMenu

export default SideNavigation
