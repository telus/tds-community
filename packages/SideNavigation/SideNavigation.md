Displays multiple links on a page for easier navigation.

### Usage Criteria

- The `SideNavigation` component is for desktop viewports only
- Use when content priority is neutral
- Use when the IA for transactional & informational experiences calls for 7+ categories
- Flexible character count for category and link titles
- We do not recommend using in conjunction with [Breadcrumbs](https://tds.telus.com/components/index.html#breadcrumbs)

```
<FlexGrid limitWidth={false} gutter={false}>
  <FlexGrid.Row>
    <FlexGrid.Col xs={5}>
      <SideNavigation accordion={true} verticalSpacing={3}>
        <SideNavigation.Link href="#">Top of the page</SideNavigation.Link>
        <SideNavigation.SubMenu label="Overview" active>
          <SideNavigation.Link href="#introduction">Introduction</SideNavigation.Link>
        </SideNavigation.SubMenu>
        <SideNavigation.SubMenu label="Reference Architecture">
          <SideNavigation.Link href="#reference">Overview</SideNavigation.Link>
        </SideNavigation.SubMenu>
      </SideNavigation>
    </FlexGrid.Col>
    <FlexGrid.Col xs={5} xsOffset={1}>
      <Box id="introduction" vertical={3}>
        <Heading level="h2">
          Overview
        </Heading>
        <Text>
          This is an introduction to the Side Navigation component
        </Text>
      </Box>
      <Box id="reference" vertical={3}>
        <Heading level="h2">
          Reference Architecture
        </Heading>
        <Text>
          Our software and technical standards are communicated through our reference architecture. The Reference Architecture serves as a source of truth for all the technical knowledge, software specifications, standards and architecture evolution.
        </Text>
      </Box>
    </FlexGrid.Col>
  </FlexGrid.Row>
</FlexGrid>
```
