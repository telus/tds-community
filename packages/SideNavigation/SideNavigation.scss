@import '~@tds/core-colours/colours';
@import '~@tds/core-responsive/responsive';

$spacing-base: 1rem; // 16px

$mobile: (
  1: $spacing-base * 0.25,
  2: $spacing-base * 0.5,
  3: $spacing-base,
  4: $spacing-base * 1.5,
  5: $spacing-base * 2,
  6: $spacing-base * 2.5,
  7: $spacing-base * 3,
  8: $spacing-base * 4
);

$desktop: (
  1: $spacing-base * 0.25,
  2: $spacing-base * 0.5,
  3: $spacing-base,
  4: $spacing-base * 2,
  5: $spacing-base * 3,
  6: $spacing-base * 4,
  7: $spacing-base * 4.5,
  8: $spacing-base * 6
);

@mixin spacingClasses ($level, $mobileValue, $desktopValue) {
  .verticalPadding-#{$level} {
    padding-top: $mobileValue;
    @include mq($from: md) {
      padding-top: $desktopValue;
    }
  }
}

@each $level, $value in $mobile {
  @include spacingClasses($level, $value, map-get($desktop, $level));
}

.container {
  position: relative;
  height: 100%;
}

.spaced {
  list-style: none;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  margin: 0;
  border-top: 1px solid $color-gainsboro;
}

.navLi {
  border-bottom: 1px solid $color-gainsboro;
  align-items: center;
}

.active {
  border-left: 4px solid $color-telus-purple;
}

li.active a {
  color: $color-telus-purple;
}

.topPosition {
  position: relative;
  max-width: inherit;
  width: 100%;
  overflow-y: auto;
}

.fixedPosition {
  position: fixed;
  max-width: inherit;
  top: 0px;
  width: inherit;
}

.bottomPosition {
  position: absolute;
  bottom: 0px;
}

.removeBottomPadding {
  padding-bottom: 0px;
}

.removeTopPadding {
  padding-top: 0px;
}
