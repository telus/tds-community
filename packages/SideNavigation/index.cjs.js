const SideNavigation = require('./dist/index.cjs')

module.exports = SideNavigation
