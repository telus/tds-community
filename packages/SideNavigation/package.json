{
  "name": "@tds/community-side-navigation",
  "version": "1.1.3",
  "description": "",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telus/tds-community.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telus/tds-community/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": "^16.8.2",
    "react-dom": "^16.8.2",
    "styled-components": "^4.1.3"
  },
  "dependencies": {
    "@tds/core-box": "^1.2.3",
    "@tds/core-colours": "^1.1.1",
    "@tds/core-decorative-icon": "^1.2.0",
    "@tds/core-text": "^1.1.0",
    "@tds/shared-animation": "1.1.0",
    "@tds/util-prop-types": "^1.3.2",
    "prop-types": "^15.6.2"
  },
  "devDependencies": {
    "@tds/core-colours": "^1.1.2",
    "@tds/shared-safe-rest": "^1.0.0"
  }
}
