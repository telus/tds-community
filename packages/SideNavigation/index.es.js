import './dist/index.css'
import SideNavigation from './dist/index.es'

export default SideNavigation
