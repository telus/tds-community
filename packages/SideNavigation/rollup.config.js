import configure from '../../config/rollup.config'
import { dependencies } from './package.json'

export default configure({
  input: './SideNavigation.jsx',
  dependencies,
})
