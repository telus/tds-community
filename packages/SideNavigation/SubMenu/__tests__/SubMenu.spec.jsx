import React from 'react'
import { mount } from 'enzyme'

import SideNavigation from '../../SideNavigation'

describe('SideNavigation.SubMenu', () => {
  const onClickFunction = jest.fn()
  const SubMenu = (
    <SideNavigation.SubMenu
      active
      label="Threefdsfdsfds"
      id="Three"
      onClick={onClickFunction}
      isOpen={false}
    >
      <SideNavigation.Link href="#">Option 1</SideNavigation.Link>
      <SideNavigation.Link href="#">Option 2</SideNavigation.Link>
      <SideNavigation.Link href="#">Option 3</SideNavigation.Link>
    </SideNavigation.SubMenu>
  )

  const SideNav = (
    <SideNavigation>
      <SideNavigation.SubMenu
        label="Threefdsfdsfds"
        id="Three"
        onClick={onClickFunction}
        isOpen={false}
      >
        <SideNavigation.Link href="#">Option 1</SideNavigation.Link>
        <SideNavigation.Link href="#">Option 2</SideNavigation.Link>
        <SideNavigation.Link href="#">Option 3</SideNavigation.Link>
      </SideNavigation.SubMenu>
    </SideNavigation>
  )

  it('renders', () => {
    const subMenu = mount(SubMenu)

    expect(subMenu).toMatchSnapshot()
  })

  it('calls onClick and internal click function', () => {
    const sideNav = mount(SideNav)

    sideNav.find('button').simulate('click')

    expect(onClickFunction).toHaveBeenCalledTimes(1)
    expect(sideNav.state('open')).toEqual(['TDS-SideNavigation-0'])

    sideNav.find('button').simulate('click')

    expect(onClickFunction).toHaveBeenCalledTimes(2)
    expect(sideNav.state('open')).toEqual([])
  })
})
