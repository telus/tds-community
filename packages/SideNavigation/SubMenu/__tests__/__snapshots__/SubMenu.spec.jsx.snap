// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavigation.SubMenu renders 1`] = `
.c4:hover {
  background-color: #f2eff4;
}

.c5 {
  color: #54595f;
}

.c5:hover {
  color: #4b286d;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  padding: 1rem 0;
  background: radial-gradient( ellipse at top, rgba(150,150,150,0.1) 0%, rgba(0,0,0,0) 70% );
}

.c0 {
  background-color: white;
  border: none;
  width: 100%;
  color: #4b286d;
  border-left: 4px solid #4b286d;
  font-family: TELUS-Web,"Helvetica Neue",Helvetica,Arial,sans-serif;
}

.c0:hover {
  background-color: #f2eff4;
  color: #4b286d;
  cursor: pointer;
}

.c1 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: left;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<SubMenu
  active={true}
  id="Three"
  isOpen={false}
  label="Threefdsfdsfds"
  onClick={[MockFunction]}
>
  <SubMenu__ButtonSubMenu
    active={true}
    aria-expanded={false}
    id="Three"
    onClick={[Function]}
  >
    <StyledComponent
      active={true}
      aria-expanded={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "SubMenu__ButtonSubMenu-sc-1p1ex8g-1",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              [Function],
            ],
          },
          "displayName": "SubMenu__ButtonSubMenu",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "SubMenu__ButtonSubMenu-sc-1p1ex8g-1",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      id="Three"
      onClick={[Function]}
    >
      <button
        aria-expanded={false}
        className="c0"
        id="Three"
        onClick={[Function]}
      >
        <SubMenu__SpaceBox
          horizontal={2}
          inline={true}
          vertical={3}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "SubMenu__SpaceBox-sc-1p1ex8g-2",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "justify-content: space-between; align-items: center; text-align: left; display: flex;",
                  ],
                },
                "displayName": "SubMenu__SpaceBox",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "SubMenu__SpaceBox-sc-1p1ex8g-2",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            horizontal={2}
            inline={true}
            vertical={3}
          >
            <Box
              className="c1"
              horizontal={2}
              inline={true}
              tag="div"
              vertical={3}
            >
              <div
                className="TDS_Box-modules__horizontalPadding-2___3hmwJ TDS_Box-modules__verticalPadding-3___Fsv37 c1"
              >
                <ColoredTextProvider
                  className={null}
                >
                  <div>
                    <Text
                      block={false}
                      bold={true}
                      invert={false}
                      size="medium"
                    >
                      <span
                        className="TDS_Typography-modules__default___KlcpP TDS_Typography-modules__medium___1rxfE TDS_Typography-modules__wordBreak___3OZx_ TDS_Typography-modules__boldFont___2YS3O TDS_Typography-modules__wordBreak___3OZx_ TDS_Typography-modules__inheritColor___SREX0"
                      >
                        Threefdsfdsfds
                      </span>
                    </Text>
                  </div>
                </ColoredTextProvider>
                <DecorativeIcon
                  size={16}
                  symbol="caretDown"
                  variant="secondary"
                >
                  <Icon
                    aria-hidden="true"
                    size={16}
                    symbol="caretDown"
                    variant="secondary"
                  >
                    <i
                      aria-hidden="true"
                      className="TDS_Icon-modules__iconCaretDown___3by8b TDS_Icon-modules__icon___13xYd TDS_Icon-modules__secondary___3Gtcp TDS_Icon-modules__size16___1pm12"
                    />
                  </Icon>
                </DecorativeIcon>
              </div>
            </Box>
          </StyledComponent>
        </SubMenu__SpaceBox>
      </button>
    </StyledComponent>
  </SubMenu__ButtonSubMenu>
  <Reveal
    duration={500}
    height={0}
    in={false}
    timeout={0}
  >
    <Transition
      appear={false}
      duration={500}
      enter={true}
      exit={true}
      in={false}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={0}
      unmountOnExit={false}
    >
      <div
        aria-hidden={true}
        data-testid="childrenContainer"
        style={
          Object {
            "height": 0,
            "overflow": "hidden",
            "transition": "height 500ms",
            "visibility": "hidden",
          }
        }
      >
        <SubMenu__SubMenuContainer>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "SubMenu__SubMenuContainer-sc-1p1ex8g-0",
                  "isStatic": true,
                  "lastClassName": "c2",
                  "rules": Array [
                    "padding: 1rem 0; background: radial-gradient(
    ellipse at top,
    rgba(150, 150, 150, 0.1) 0%,
    rgba(0, 0, 0, 0) 70%
  );",
                  ],
                },
                "displayName": "SubMenu__SubMenuContainer",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "SubMenu__SubMenuContainer-sc-1p1ex8g-0",
                "target": "ul",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={[Function]}
          >
            <ul
              className="c2"
            >
              <li
                key=".0"
              >
                <Link
                  active={false}
                  href="#"
                  reactRouterLinkComponent={null}
                  subMenuLink={true}
                  to={null}
                >
                  <Link__StyledAnchor
                    href="#"
                    to={null}
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "Link__StyledAnchor-gur8v7-2",
                            "isStatic": true,
                            "lastClassName": "c3",
                            "rules": Array [
                              "text-decoration: none;",
                            ],
                          },
                          "displayName": "Link__StyledAnchor",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "Link__StyledAnchor-gur8v7-2",
                          "target": "a",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      href="#"
                      to={null}
                    >
                      <a
                        className="c3"
                        href="#"
                        to={null}
                      >
                        <Link__BoxContainer
                          active={0}
                          horizontal={3}
                          vertical={2}
                        >
                          <StyledComponent
                            active={0}
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "Link__BoxContainer-gur8v7-0",
                                  "isStatic": false,
                                  "lastClassName": "c4",
                                  "rules": Array [
                                    [Function],
                                  ],
                                },
                                "displayName": "Link__BoxContainer",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "Link__BoxContainer-gur8v7-0",
                                "target": [Function],
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            horizontal={3}
                            vertical={2}
                          >
                            <Box
                              active={0}
                              className="c4"
                              horizontal={3}
                              inline={false}
                              tag="div"
                              vertical={2}
                            >
                              <div
                                active={0}
                                className="TDS_Box-modules__horizontalPadding-3___2uoUp TDS_Box-modules__verticalPadding-2___1Uh2T c4"
                              >
                                <Link__StyledTextProvider
                                  active={false}
                                >
                                  <StyledComponent
                                    active={false}
                                    forwardedComponent={
                                      Object {
                                        "$$typeof": Symbol(react.forward_ref),
                                        "attrs": Array [],
                                        "componentStyle": ComponentStyle {
                                          "componentId": "Link__StyledTextProvider-gur8v7-1",
                                          "isStatic": false,
                                          "lastClassName": "c5",
                                          "rules": Array [
                                            [Function],
                                          ],
                                        },
                                        "displayName": "Link__StyledTextProvider",
                                        "foldedComponentIds": Array [],
                                        "render": [Function],
                                        "styledComponentId": "Link__StyledTextProvider-gur8v7-1",
                                        "target": [Function],
                                        "toString": [Function],
                                        "warnTooManyClasses": [Function],
                                        "withComponent": [Function],
                                      }
                                    }
                                    forwardedRef={null}
                                  >
                                    <ColoredTextProvider
                                      active={false}
                                      className="c5"
                                    >
                                      <div
                                        className="c5"
                                      >
                                        <Text
                                          block={false}
                                          bold={false}
                                          invert={false}
                                          size="small"
                                        >
                                          <span
                                            className="TDS_Typography-modules__default___KlcpP TDS_Typography-modules__small___mt5lS TDS_Typography-modules__wordBreak___3OZx_ TDS_Typography-modules__smallFont___3S6nz TDS_Typography-modules__inheritColor___SREX0"
                                          >
                                            Option 1
                                          </span>
                                        </Text>
                                      </div>
                                    </ColoredTextProvider>
                                  </StyledComponent>
                                </Link__StyledTextProvider>
                              </div>
                            </Box>
                          </StyledComponent>
                        </Link__BoxContainer>
                      </a>
                    </StyledComponent>
                  </Link__StyledAnchor>
                </Link>
              </li>
              <li
                key=".1"
              >
                <Link
                  active={false}
                  href="#"
                  reactRouterLinkComponent={null}
                  subMenuLink={true}
                  to={null}
                >
                  <Link__StyledAnchor
                    href="#"
                    to={null}
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "Link__StyledAnchor-gur8v7-2",
                            "isStatic": true,
                            "lastClassName": "c3",
                            "rules": Array [
                              "text-decoration: none;",
                            ],
                          },
                          "displayName": "Link__StyledAnchor",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "Link__StyledAnchor-gur8v7-2",
                          "target": "a",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      href="#"
                      to={null}
                    >
                      <a
                        className="c3"
                        href="#"
                        to={null}
                      >
                        <Link__BoxContainer
                          active={0}
                          horizontal={3}
                          vertical={2}
                        >
                          <StyledComponent
                            active={0}
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "Link__BoxContainer-gur8v7-0",
                                  "isStatic": false,
                                  "lastClassName": "c4",
                                  "rules": Array [
                                    [Function],
                                  ],
                                },
                                "displayName": "Link__BoxContainer",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "Link__BoxContainer-gur8v7-0",
                                "target": [Function],
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            horizontal={3}
                            vertical={2}
                          >
                            <Box
                              active={0}
                              className="c4"
                              horizontal={3}
                              inline={false}
                              tag="div"
                              vertical={2}
                            >
                              <div
                                active={0}
                                className="TDS_Box-modules__horizontalPadding-3___2uoUp TDS_Box-modules__verticalPadding-2___1Uh2T c4"
                              >
                                <Link__StyledTextProvider
                                  active={false}
                                >
                                  <StyledComponent
                                    active={false}
                                    forwardedComponent={
                                      Object {
                                        "$$typeof": Symbol(react.forward_ref),
                                        "attrs": Array [],
                                        "componentStyle": ComponentStyle {
                                          "componentId": "Link__StyledTextProvider-gur8v7-1",
                                          "isStatic": false,
                                          "lastClassName": "c5",
                                          "rules": Array [
                                            [Function],
                                          ],
                                        },
                                        "displayName": "Link__StyledTextProvider",
                                        "foldedComponentIds": Array [],
                                        "render": [Function],
                                        "styledComponentId": "Link__StyledTextProvider-gur8v7-1",
                                        "target": [Function],
                                        "toString": [Function],
                                        "warnTooManyClasses": [Function],
                                        "withComponent": [Function],
                                      }
                                    }
                                    forwardedRef={null}
                                  >
                                    <ColoredTextProvider
                                      active={false}
                                      className="c5"
                                    >
                                      <div
                                        className="c5"
                                      >
                                        <Text
                                          block={false}
                                          bold={false}
                                          invert={false}
                                          size="small"
                                        >
                                          <span
                                            className="TDS_Typography-modules__default___KlcpP TDS_Typography-modules__small___mt5lS TDS_Typography-modules__wordBreak___3OZx_ TDS_Typography-modules__smallFont___3S6nz TDS_Typography-modules__inheritColor___SREX0"
                                          >
                                            Option 2
                                          </span>
                                        </Text>
                                      </div>
                                    </ColoredTextProvider>
                                  </StyledComponent>
                                </Link__StyledTextProvider>
                              </div>
                            </Box>
                          </StyledComponent>
                        </Link__BoxContainer>
                      </a>
                    </StyledComponent>
                  </Link__StyledAnchor>
                </Link>
              </li>
              <li
                key=".2"
              >
                <Link
                  active={false}
                  href="#"
                  reactRouterLinkComponent={null}
                  subMenuLink={true}
                  to={null}
                >
                  <Link__StyledAnchor
                    href="#"
                    to={null}
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "Link__StyledAnchor-gur8v7-2",
                            "isStatic": true,
                            "lastClassName": "c3",
                            "rules": Array [
                              "text-decoration: none;",
                            ],
                          },
                          "displayName": "Link__StyledAnchor",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "Link__StyledAnchor-gur8v7-2",
                          "target": "a",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      href="#"
                      to={null}
                    >
                      <a
                        className="c3"
                        href="#"
                        to={null}
                      >
                        <Link__BoxContainer
                          active={0}
                          horizontal={3}
                          vertical={2}
                        >
                          <StyledComponent
                            active={0}
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "Link__BoxContainer-gur8v7-0",
                                  "isStatic": false,
                                  "lastClassName": "c4",
                                  "rules": Array [
                                    [Function],
                                  ],
                                },
                                "displayName": "Link__BoxContainer",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "Link__BoxContainer-gur8v7-0",
                                "target": [Function],
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            horizontal={3}
                            vertical={2}
                          >
                            <Box
                              active={0}
                              className="c4"
                              horizontal={3}
                              inline={false}
                              tag="div"
                              vertical={2}
                            >
                              <div
                                active={0}
                                className="TDS_Box-modules__horizontalPadding-3___2uoUp TDS_Box-modules__verticalPadding-2___1Uh2T c4"
                              >
                                <Link__StyledTextProvider
                                  active={false}
                                >
                                  <StyledComponent
                                    active={false}
                                    forwardedComponent={
                                      Object {
                                        "$$typeof": Symbol(react.forward_ref),
                                        "attrs": Array [],
                                        "componentStyle": ComponentStyle {
                                          "componentId": "Link__StyledTextProvider-gur8v7-1",
                                          "isStatic": false,
                                          "lastClassName": "c5",
                                          "rules": Array [
                                            [Function],
                                          ],
                                        },
                                        "displayName": "Link__StyledTextProvider",
                                        "foldedComponentIds": Array [],
                                        "render": [Function],
                                        "styledComponentId": "Link__StyledTextProvider-gur8v7-1",
                                        "target": [Function],
                                        "toString": [Function],
                                        "warnTooManyClasses": [Function],
                                        "withComponent": [Function],
                                      }
                                    }
                                    forwardedRef={null}
                                  >
                                    <ColoredTextProvider
                                      active={false}
                                      className="c5"
                                    >
                                      <div
                                        className="c5"
                                      >
                                        <Text
                                          block={false}
                                          bold={false}
                                          invert={false}
                                          size="small"
                                        >
                                          <span
                                            className="TDS_Typography-modules__default___KlcpP TDS_Typography-modules__small___mt5lS TDS_Typography-modules__wordBreak___3OZx_ TDS_Typography-modules__smallFont___3S6nz TDS_Typography-modules__inheritColor___SREX0"
                                          >
                                            Option 3
                                          </span>
                                        </Text>
                                      </div>
                                    </ColoredTextProvider>
                                  </StyledComponent>
                                </Link__StyledTextProvider>
                              </div>
                            </Box>
                          </StyledComponent>
                        </Link__BoxContainer>
                      </a>
                    </StyledComponent>
                  </Link__StyledAnchor>
                </Link>
              </li>
            </ul>
          </StyledComponent>
        </SubMenu__SubMenuContainer>
      </div>
    </Transition>
  </Reveal>
</SubMenu>
`;
